resources/views/layouts/navbar.blade.php

```diff
@@ -19,23 +19,36 @@
             <b-alert v-for="(item, index) in alerts" :key="index" class="d-none d-lg-block alertBox" :show="item.alertShow" :variant="item.alertVariant" dismissible fade @dismissed="alertDismissed(item)" @dismiss-count-down="alertDownChanged($event, item)" style="white-space:pre-line">@{{item.alertText}}</b-alert>
         </div>
 
+        @php
+            $menuItems = [];
+            $existsMenuProvider = Menu::get('customtopnav') !== null;
+            $customNav = $existsMenuProvider ? Menu::get('customtopnav')->items->all() : [];
+            $defaultNav = Menu::get('topnav')->items->all();
+            foreach(array_merge($customNav, $defaultNav) as $item) {
+                $newItem = (array) $item;
+                $newItem['link'] = $item->url();
+                $newItem['isCustom'] = in_array($item, $customNav);
+                $menuItems[] = $newItem;
+            }
+            // If a menu provider is installed, remove menu items from ProcessMaker but preserve any other (from packages, for example)
+            if ($existsMenuProvider) {
+                $menuItems = array_filter($menuItems, function ($item) use($customNav) {
+                    $itemRoute = Route::getRoutes()->match(Request::create($item['link']));
+                    $isCoreLink =  !$itemRoute->isFallBack && isset($itemRoute->action['controller']) && strpos($itemRoute->action['controller'], "ProcessMaker\\Http\\") === 0;
+                    return !$isCoreLink || $item['isCustom'];
+                });
+            }
+        @endphp
+
         <b-navbar-nav class="d-flex align-items-center">
-            @foreach(Menu::get('topnav')->items as $item)
-                <b-nav-item href="{{ $item->url() }}" {{$item->isActive !== false ? 'active': ''}}>
-                    {!! $item->title !!}
+                <b-nav-item v-for="item in {{ json_encode ($menuItems) }}"
```

Comment:
Move the menuItems to a data element in the javascript. The menu items shouldn't be processed on each page load.

Like this:

<script>
export default {
  data() {
    return {
      menuitems: {{ json_encode ($menuItems) }}
    }
  },

Reply:
The backend logic was moved to a manager and the JS variable is set in the layout.